src/v/raft/tests/basic_raft_fixture_test.cc

```diff
@@ -862,3 +862,108 @@ TEST_F_CORO(raft_fixture, test_no_stepdown_on_append_entries_timeout) {
     ASSERT_EQ_CORO(term_before, new_leader_node.raft()->term());
     ASSERT_TRUE_CORO(new_leader_node.raft()->is_leader());
 }
+
+/**
+ * This synthetic test is there to trigger a situation in which follower
+ * receives an append entries request which contains only batches that matches
+ * its log. This trigger a condition in which the follower should reply with
+ * success to the leader so the leader can continue recovery process.
+ *
+ * The test uses reply interception to 'trick' the leader to send the append
+ * entries with the batches that the follower already has.
+ */
+TEST_F_CORO(raft_fixture, test_redelivery_of_matching_logs) {
+    co_await create_simple_group(3);
+    auto leader_id = co_await wait_for_leader(10s);
+    model::node_id non_leader_id{0};
+    for (auto& [id, n] : nodes()) {
+        n->set_default_recovery_read_size(1);
+    }
+
+    for (auto& [id, _] : nodes()) {
+        if (id != leader_id) {
+            non_leader_id = id;
+            break;
+        }
+    }
+    auto& leader_node = node(leader_id);
+    /**
+     * Replicate data to all nodes
+     */
+    auto r = co_await leader_node.raft()->replicate(
+      make_batches(200, 1, 10),
+      replicate_options(consistency_level::quorum_ack, 10s));
+    /**
+     * Prevent one node from receiving append entries
+     */
+    leader_node.on_dispatch([non_leader_id](model::node_id id, raft::msg_type) {
+        if (id == non_leader_id) {
+            throw std::runtime_error("error");
+        }
+        return ss::now();
+    });
+    /**
+     * Replicate data with one of the nodes being blocked
+     */
+    r = co_await leader_node.raft()->replicate(
+      make_batches(200, 1, 10),
+      replicate_options(consistency_level::quorum_ack, 10s));
+    ASSERT_FALSE_CORO(r.has_error());
+    auto term_1_match_offset = node(non_leader_id).raft()->dirty_offset();
+    /**
+     * Append some batches to the blocked node with different term. This is a
+     * 'brute force' solution to generate a situation in which the follower
+     * and leader log would diverge. This simulates a situation in which the
+     * blocked follower replica was a leader in term 2 and accepted some writes
```

Comment:
> This simulates a situation in which the blocked follower replica was a leader in term 2

... which it could not be elected for. If another node (term 1 leader or third node) voted for it this voter would have bumped its term to 2 when having an empty log. It would not be able to save term 1 data then.